commit d5ca4679a5ebe9deebf07a939ddd472c4f71b97e (refs/changes/47/2547/1)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-09-12T14:57:35+00:00 (2 years, 1 month ago)
    
    MORAY-431 Errors after query has timed out are emitted when there are no "error" listeners

diff --git a/test/sql.test.js b/test/sql.test.js
index b53784c..849b1ee 100644
--- a/test/sql.test.js
+++ b/test/sql.test.js
@@ -119,3 +119,21 @@ test('sql - client timeout respected', function (t) {
         t.end();
     });
 });
+
+test('sql - error after timeout', function (t) {
+    q = c.sql('SELECT pg_sleep(5); SELECT * FROM nonexistent;',
+        [], { timeout: 1 });
+    q.on('error', function (err) {
+        t.ok(VError.hasCauseWithName(err, 'QueryTimeoutError'),
+            'correct error');
+        t.end();
+    });
+    q.on('record', function (r) {
+        t.fail('no row should be returned');
+        t.deepEqual(r, null, 'no row should be returned - value');
+    });
+    q.on('end', function (r) {
+        t.fail('"error" should have been emitted');
+        t.end();
+    });
+});
